### PR TITLE
MLIRConverter: cleanup and add tests

### LINKED ISSUE #33: Add a clean interface to the MLIR converter
Currently the `mlir_converter.py` is a mess. When we want to use it from the ChocoPy project, we have to rewrite parts of it to make it possible to wrap it into a tool.

### PR DESCRIPTION
**Pull request by Dinistro**
This somewhat fixes #33 

In the ChocoPy case we will probably call the converter from `choco-opt`, thus writing a tool for it is not necessary I think. A tool might only be helpful for writing filecheck tests, but these will not work in the CI due to missing MLIR as a dependency.

### CODE REVIEW

**Comment by Dinistro** [comment]
If you have the python bindings from llvm/main installed on your machine, you can test this locally, otherwise it will not work.

File: src/xdsl/mlir_converter.py [hunk 1]
```diff
@@ -1,33 +1,7 @@
...
-  %3 : !index = memref.load(%1 : !memref<[1 : !i64], !index>, %zero : !index) 
-  %4 : !memref<[10 : !i64, 2 : !i64], !index> = memref.alloc() ["alignment" = 0 : !i64, "operand_segment_sizes" = !dense<!vector<[2 : !i64], !i32>, [0 : !i32, 0 : !i32]>]
-  memref.store(%2 : !index, %4 : !memref<[10 : !i64, 2 : !i64], !index>, %2 : !index, %3 : !index) 
-  memref.dealloc(%1 : !memref<[1 : !i64], !index>) 
-  memref.dealloc(%4 : !memref<[10 : !i64, 2 : !i64], !index>) 
-  std.return()
-}
-}
-"""
+from xdsl.dialects.memref import MemRefType
```
Comment (Dinistro) [inline src/xdsl/mlir_converter.py:4]: This is currently required because we define `MemRefType` outside of builtin.

**Review by webmiche — APPROVED** [review]

**Comment by Dinistro** [comment]
@math-fehr Do you think that we can merge this?

**Comment by tobiasgrosser** [comment]
Not sure if @math-fehr is available tonight, as he has family visiting. As the patch looks pretty simple and you have the OK of @webmiche, you might be able to go ahead nevertheless, in case this is blocking you. I can update the xdsl package if this is needed.

**Comment by math-fehr** [comment]
I quickly read through it and it seems good!

**Comment by Dinistro** [comment]
Thank you guys. Sorry to interrupt your family visit. 

@tobiasgrosser If you could publish a new xdsl package version, I would be glad. I'm assuming that I do not have permissions to do so.

**Pull request merged**